### PR TITLE
Enemy attack gamestate

### PR DESCRIPTION
**Pull request by Dududungha**

### CODE REVIEW

File: client/src/components/PlayerComponent.cpp [hunk 1]
```diff
@@ -3,9 +3,16 @@
...
+#include <Health.hpp>
 
 void PlayerComponent::Update(float deltaTime) {
-    if (glm::length(owner->GetComponent<Mover>()->movementHeading) > 0) {
+    if (owner->GetComponent<Health>()->GetHealth() <= 0) {
+        // TODO: add animation for death
+        // TODO: stop player movement, probably just set speed in mover.
+        owner->GetComponent<NetTransform>()->SetRotation(glm::vec3(90.0f, 0.0f, 0.0f));
+        //owner->GetComponent<AnimationPlayer>()->play("dead");
+        owner->GetComponent<AnimationPlayer>()->play("idle");
```
Comment (Kikit24) [inline client/src/components/PlayerComponent.cpp:14]: Player death animation is called “dying"

**Review by junweic01 — COMMENTED** [review]
Looks good to me!

File: server/prefabs/AttackManager.cpp [hunk 2]
```diff
@@ -47,6 +48,15 @@ void AttackManager::newSwipeAttack() {
 }
 
 void AttackManager::update(float deltaTime) {
+    // this is such a wacky way to keep track of number of players alive rn
+    int numAlive = 0;
```
Comment (Kikit24) [inline server/prefabs/AttackManager.cpp:52]: Why isn’t this in NetworkManager.update() since numAlive is stored there?

**Review by Kikit24 — CHANGES_REQUESTED** [review]
Looks good, thanks for the refactoring + logic updates! Just a few comments

File: server/prefabs/AttackManager.cpp (hunk 2 above)
Comment (Kikit24) [reply server/prefabs/AttackManager.cpp:52]: resolved: AttackManager needed to access player list.

**Pull request merged**